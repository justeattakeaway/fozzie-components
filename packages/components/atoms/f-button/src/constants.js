// eslint-disable-next-line import/prefer-default-export
export const VALID_BUTTON_TYPES = {
    iconButton: ['primary', 'secondary', 'ghost', 'ghostTertiary'],
    button: ['primary', 'secondary', 'outline', 'ghost', 'link']
};

export const VALID_BUTTON_SIZES = ['xsmall', 'small', 'medium', 'large'];
