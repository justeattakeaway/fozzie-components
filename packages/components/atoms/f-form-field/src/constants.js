export const VALID_INPUT_TYPES = ['text', 'email', 'password', 'radio', 'checkbox', 'number'];
export const CUSTOM_INPUT_TYPES = ['dropdown'];
export const DEFAULT_INPUT_TYPE = 'text';

export const VALID_LABEL_STYLES = ['default', 'inline', 'inlineNarrow'];

export const MOBILE_WIDTH = 768;
