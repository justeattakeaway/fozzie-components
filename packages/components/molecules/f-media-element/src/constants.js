export const ALIGN = {
    LEFT: 'left',
    RIGHT: 'right',
    CENTER: 'center'
};

export const FONT_SIZE = {
    SM: 'sm',
    MD: 'md',
    LG: 'lg',
    XL: 'xl',
    XXL: 'xxl'
};

export const MODIFIER_RULES_MAP = {
    '>=': 'gte',
    '>': 'gt',
    '<': 'lt',
    '<=': 'lte'
};

export const MODIFIER_OPPOSITES_RULES_MAP = {
    '>=': 'lt',
    '>': 'lte',
    '<': 'gte',
    '<=': 'gt'
};
