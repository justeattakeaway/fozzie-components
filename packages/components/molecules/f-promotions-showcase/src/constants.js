export const STYLE_TEXT = 'styleText';
export const STYLE_EMPHASIZED = 'styleEmphasized';
export const STYLE_LINK = 'styleLink';

export default {
    STYLE_EMPHASIZED,
    STYLE_LINK,
    STYLE_TEXT
};
