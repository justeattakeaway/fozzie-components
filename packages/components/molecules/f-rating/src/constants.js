// eslint-disable-next-line import/prefer-default-export
export const VALID_STAR_RATING_SIZES = ['xsmall', 'small', 'medium', 'large'];
export const VALID_STAR_RATING_DISPLAY_TYPE = ['short', 'medium', 'long'];
export const VALID_STAR_FONT_SIZES = ['default', 'large'];
