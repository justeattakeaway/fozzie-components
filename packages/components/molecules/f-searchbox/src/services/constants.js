export const LOCATION_COOKIE_PROPS = [
    'unitNumber',
    'street',
    'houseNo',
    'city',
    'sublocality',
    'state'
];
export const JE_LOCATION = 'je-location';

export const JE_FULL_ADDRESS_DETAILS = 'je-full-address-details';

export const VUEX_MODULE_NAMESPACE = 'fSearchboxModule';
