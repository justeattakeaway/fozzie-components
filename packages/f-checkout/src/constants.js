export const CHECKOUT_METHOD_COLLECTION = 'Collection';
export const CHECKOUT_METHOD_DELIVERY = 'Delivery';

export const VALID_CHECKOUT_METHOD = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];
