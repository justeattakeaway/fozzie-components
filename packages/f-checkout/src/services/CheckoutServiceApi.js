import axios from 'axios';

export default {
    async submitCheckout (url, tenant, data, timeout) {
        const config = {
            method: 'post',
            headers: {
                'Content-Type': 'application/json',
                'Accept-Tenant': tenant
            },
            timeout
        };
        return axios
            .post(url, data, config);
    },

    async getCheckout (url, tenant, timeout) {
        const config = {
            method: 'get',
            headers: {
                'Content-Type': 'application/json',
                'Accept-Tenant': tenant
            },
            timeout
        };
        return axios
            .get(url, config);
    }
};
