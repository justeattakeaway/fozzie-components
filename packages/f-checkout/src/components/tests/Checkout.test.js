import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import { VALID_CHECKOUT_METHOD, CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_COLLECTION } from '../../constants';
import VueCheckout from '../Checkout.vue';
import CheckoutServiceApi from '../../services/CheckoutServiceApi';
import EventNames from '../../event-names';
import tenantConfigs from '../../tenants';
import CheckoutGetMockData from '../../demo/checkout-delivery.json';

jest.mock('../../services/CheckoutServiceApi', () => ({ submitCheckout: jest.fn(), getCheckout: jest.fn() }));

const localVue = createLocalVue();

localVue.use(VueI18n);

const defaultCheckoutGetData = {
    customer: {
        firstName: '',
        phoneNumber: null
    },
    fulfillment: {
        address: {
            lines: [null, null, null, null],
            postalCode: null
        }
    },
    serviceType: CHECKOUT_METHOD_DELIVERY
};

const i18n = {
    locale: 'en-GB',
    messages: tenantConfigs['en-GB']
};

describe('Checkout', () => {
    allure.feature('Checkout');
    const checkoutUrl = 'http://localhost/account/register';

    it('should be defined', () => {
        const propsData = { checkoutUrl };

        const wrapper = shallowMount(VueCheckout, {
            i18n,
            localVue,
            propsData
        });

        expect(wrapper.exists()).toBe(true);
    });

    describe('data ::', () => {
        describe('serviceType ::', () => {
            it.each(VALID_CHECKOUT_METHOD)('should update the Selector `ordermethod` attribute to match serviceType=%p', async definedType => {
                // Arrange
                const propsData = {
                    checkoutUrl
                };

                const data = {
                    serviceType: definedType
                };

                // Act
                const wrapper = shallowMount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });

                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                const selectorComponent = wrapper.find('[data-test-id="selector"]');

                // Assert
                expect(selectorComponent.attributes('ordermethod')).toEqual(definedType);
            });

            it('should display the address block if set to `delivery`', async () => {
                // Arrange
                const propsData = {
                    checkoutUrl
                };

                const data = {
                    serviceType: CHECKOUT_METHOD_DELIVERY
                };

                // Act
                const wrapper = shallowMount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });

                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                const addressBlock = wrapper.find('[data-test-id="address-block"]');

                // Assert
                expect(addressBlock.exists()).toBe(true);
            });

            it('should not display the address block if set to `collection`', async () => {
                // Arrange
                const propsData = {
                    checkoutUrl
                };

                const data = {
                    serviceType: CHECKOUT_METHOD_COLLECTION
                };

                // Act
                const wrapper = shallowMount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });

                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                const addressBlock = wrapper.find('[data-test-id="address-block"]');

                // Assert
                expect(addressBlock.exists()).toBe(false);
            });
        });
    });

    describe('computed ::', () => {
        const propsData = { checkoutUrl };
        const data = { firstName: 'name' };

        describe('name ::', () => {
            it('should capitalize `firstName` data', async () => {
                // Arrange
                const wrapper = shallowMount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });

                const name = wrapper.find("[data-test-id='checkout-card-component']");

                // Act
                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                // Assert
                expect(name.props('cardHeading')).toContain('Name');
            });
        });

        describe('title ::', () => {
            it('should add `name` to title text', async () => {
                // Arrange
                const wrapper = shallowMount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });

                const name = wrapper.find("[data-test-id='checkout-card-component']");

                // Act
                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                // Assert
                expect(name.props('cardHeading')).toEqual('Name, confirm your details');
            });
        });
    });

    describe('when form submitted', () => {
        const mobileNumber = '07777777777';

        const address = {
            line1: 'Address First Line',
            city: 'City',
            postcode: 'EE1E 1EE'
        };

        describe('if serviceType set to `collection`', () => {
            const propsData = {
                checkoutUrl
            };

            let wrapper;

            beforeEach(() => {
                CheckoutServiceApi.submitCheckout.mockImplementation(async () => Promise.resolve());
                CheckoutServiceApi.getCheckout.mockImplementation(async () => Promise.resolve({ data: { ...defaultCheckoutGetData, serviceType: CHECKOUT_METHOD_COLLECTION } }));

                wrapper = mount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });
            });

            afterEach(() => {
                CheckoutServiceApi.submitCheckout.mockClear();
                CheckoutServiceApi.getCheckout.mockClear();
            });

            it('should emit success event when all fields are populated correctly', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-mobile-number"]').setValue(mobileNumber);

                // Act
                await wrapper.vm.onFormSubmit();

                // Assert
                expect(wrapper.emitted(EventNames.CheckoutSuccess).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)).toBeUndefined();
            });

            it('should show error message and emit failure event when the mobile number field is not populated', async () => {
                // Arrange && Act
                await wrapper.vm.onFormSubmit();
                const mobileNumberEmptyMessage = wrapper.find('[data-test-id="error-mobile-number"]');

                // Assert
                expect(wrapper.vm.isMobileNumberValid).toBe(false);
                expect(mobileNumberEmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('mobileNumber');
            });

            it('should show error message and emit failure event when the mobile number field is populated with a < 10 numbers', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-mobile-number"]').setValue('077777');

                // Act
                await wrapper.vm.onFormSubmit();
                const mobileNumberEmptyMessage = wrapper.find('[data-test-id="error-mobile-number"]');

                // Assert
                expect(wrapper.vm.isMobileNumberValid).toBe(false);
                expect(mobileNumberEmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('mobileNumber');
            });

            it('should show error message and emit failure event when the mobile number field is populated with non numeric value', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-mobile-number"]').setValue('hs;-j`$e&1l');

                // Act
                await wrapper.vm.onFormSubmit();
                const mobileNumberEmptyMessage = wrapper.find('[data-test-id="error-mobile-number"]');

                // Assert
                expect(wrapper.vm.isMobileNumberValid).toBe(false);
                expect(mobileNumberEmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('mobileNumber');
            });

            it('should not show error message or emit failure event when the address input fields are not populated', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-mobile-number"]').setValue(mobileNumber);

                // Act
                await wrapper.vm.onFormSubmit();

                // Assert
                expect(wrapper.emitted(EventNames.CheckoutSuccess).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)).toBeUndefined();
            });

            it('should not create validations for address', () => {
                // Assert
                expect(wrapper.vm.$v.address).toBeUndefined();
            });
        });

        describe('if serviceType set to `delivery`', () => {
            const propsData = {
                checkoutUrl
            };

            let wrapper;

            beforeEach(() => {
                CheckoutServiceApi.submitCheckout.mockImplementation(async () => Promise.resolve());
                CheckoutServiceApi.getCheckout.mockImplementation(async () => Promise.resolve({ data: { ...defaultCheckoutGetData, serviceType: CHECKOUT_METHOD_DELIVERY } }));

                wrapper = mount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });
            });

            afterEach(() => {
                CheckoutServiceApi.submitCheckout.mockClear();
                CheckoutServiceApi.getCheckout.mockClear();
            });

            it('should emit success event when all fields are populated correctly', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-mobile-number"]').setValue(mobileNumber);
                wrapper.find('[data-test-id="input-address-line-1"]').setValue(address.line1);
                wrapper.find('[data-test-id="input-address-city"]').setValue(address.city);
                wrapper.find('[data-test-id="input-address-postcode"]').setValue(address.postcode);

                // Act
                await wrapper.vm.onFormSubmit();

                // Assert
                expect(wrapper.emitted(EventNames.CheckoutSuccess).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)).toBeUndefined();
            });

            it('should emit failure event and display error message when address line1 input field is empty', async () => {
                // Arrange && Act
                await wrapper.vm.onFormSubmit();
                const addressLine1EmptyMessage = wrapper.find('[data-test-id="error-address-line1-empty"]');

                // Assert
                expect(addressLine1EmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('address');
            });

            it('should emit failure event and display error message when city input field is empty', async () => {
                // Arrange && Act
                await wrapper.vm.onFormSubmit();
                const addressCityEmptyMessage = wrapper.find('[data-test-id="error-address-city-empty"]');

                // Assert
                expect(addressCityEmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('address');
            });

            it('should emit failure event and display error message when postcode input field is empty', async () => {
                // Arrange && Act
                await wrapper.vm.onFormSubmit();
                const addressPostcodeEmptyMessage = wrapper.find('[data-test-id="error-address-postcode-empty"]');

                // Assert
                expect(addressPostcodeEmptyMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('address');
            });

            it('should emit failure event and display error message when postcode contains incorrect characters', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-address-postcode"]').setValue('?!hdb-se');

                // Act
                await wrapper.vm.onFormSubmit();
                const addressPostcodeTypeErrorMessage = wrapper.find('[data-test-id="error-address-postcode-type-error"]');

                // Assert
                expect(addressPostcodeTypeErrorMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('address');
            });


            it('should emit failure event and display error message when postcode contains incorrect characters', async () => {
                // Arrange
                wrapper.find('[data-test-id="input-address-postcode"]').setValue('EC4M 7R');

                // Act
                await wrapper.vm.onFormSubmit();
                const addressPostcodeTypeErrorMessage = wrapper.find('[data-test-id="error-address-postcode-type-error"]');

                // Assert
                expect(addressPostcodeTypeErrorMessage).toMatchSnapshot();
                expect(wrapper.emitted(EventNames.CheckoutFailure).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutFailure)[0][0].invalidFields).toContain('address');
            });

            it('should create validations for address', () => {
                // Assert
                expect(wrapper.vm.$v.address.line1).toBeDefined();
                expect(wrapper.vm.$v.address.city).toBeDefined();
                expect(wrapper.vm.$v.address.postcode).toBeDefined();
            });
        });
    });

    describe('when form is loaded', () => {
        const propsData = {
            checkoutUrl
        };

        describe('when request fails', () => {
            let wrapper;

            beforeEach(() => {
                CheckoutServiceApi.getCheckout.mockImplementation(async () => Promise.reject());

                wrapper = mount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });
            });

            afterEach(() => {
                CheckoutServiceApi.getCheckout.mockClear();
            });

            it('should emit failure event', async () => {
                expect(wrapper.emitted(EventNames.CheckoutGetFailure).length).toBe(1);
            });
        });

        describe('when request succeeds', () => {
            let wrapper;

            beforeEach(() => {
                CheckoutServiceApi.getCheckout.mockImplementation(async () => Promise.resolve({ data: CheckoutGetMockData }));

                wrapper = mount(VueCheckout, {
                    i18n,
                    localVue,
                    propsData
                });
            });

            afterEach(() => {
                CheckoutServiceApi.getCheckout.mockClear();
            });

            it('should emit success event', async () => {
                expect(wrapper.emitted(EventNames.CheckoutGetSuccess).length).toBe(1);
                expect(wrapper.emitted(EventNames.CheckoutGetFailure)).toBeUndefined();
            });

            it('should set mobile number', async () => {
                expect(wrapper.find('[data-test-id="input-mobile-number"]').element.value).toBe('+447111111111');
            });

            it('should set address fields', async () => {
                expect(wrapper.find('[data-test-id="input-address-line-1"]').element.value).toBe('1 Bristol Road');
                expect(wrapper.find('[data-test-id="input-address-line-2"]').element.value).toBe('Flat 1');
                expect(wrapper.find('[data-test-id="input-address-city"]').element.value).toBe('Bristol');
                expect(wrapper.find('[data-test-id="input-address-postcode"]').element.value).toBe('BS1 1AA');
            });
        });
    });
});
