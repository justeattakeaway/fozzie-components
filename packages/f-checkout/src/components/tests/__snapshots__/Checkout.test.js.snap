// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout when form submitted if serviceType set to \`collection\` should show error message and emit failure event when the mobile number field is not populated 1`] = `
<p data-test-id="error-mobile-number" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                        Your phone number should be at least 10 characters long and shouldn't contain letters or special characters
                    </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`collection\` should show error message and emit failure event when the mobile number field is populated with a < 10 numbers 1`] = `
<p data-test-id="error-mobile-number" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                        Your phone number should be at least 10 characters long and shouldn't contain letters or special characters
                    </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`collection\` should show error message and emit failure event when the mobile number field is populated with non numeric value 1`] = `
<p data-test-id="error-mobile-number" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                        Your phone number should be at least 10 characters long and shouldn't contain letters or special characters
                    </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`delivery\` should emit failure event and display error message when address line1 input field is empty 1`] = `
<p data-test-id="error-address-line1-empty" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                    Please enter the first line of your address
                </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`delivery\` should emit failure event and display error message when city input field is empty 1`] = `
<p data-test-id="error-address-city-empty" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                    Please enter your town or city
                </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`delivery\` should emit failure event and display error message when postcode contains incorrect characters 1`] = `
<p data-test-id="error-address-postcode-type-error" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                This doesn't look like a UK postcode, can you enter it again please?
            </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`delivery\` should emit failure event and display error message when postcode contains incorrect characters 2`] = `
<p data-test-id="error-address-postcode-type-error" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                This doesn't look like a UK postcode, can you enter it again please?
            </span></p>
`;

exports[`Checkout when form submitted if serviceType set to \`delivery\` should emit failure event and display error message when postcode input field is empty 1`] = `
<p data-test-id="error-address-postcode-empty" data-test-title="" class="ErrorMessage_c-errorMessage__5fre"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 12" class="c-ficon c-ficon--warning ErrorMessage_c-errorMessage-icon_3E2NJ">
    <path d="M6.5 4.24a.32.32 0 0 1 .66 0v2.6a.32.32 0 0 1-.65 0v-2.6zm.33 4.88a.65.65 0 1 1 0-1.3.65.65 0 0 1 0 1.3zm6.36 1.24a1.3 1.3 0 0 1-1.12.7H1.6a1.29 1.29 0 0 1-1.04-2L5.8 1.22a1.25 1.25 0 0 1 2.08 0l5.24 7.82c.27.4.3.9.08 1.31zm-.62-.94L7.33 1.59a.6.6 0 0 0-1 0L1.1 9.42a.64.64 0 0 0 .5 1h10.47a.64.64 0 0 0 .51-1z" fill="#D50525"></path>
  </svg><span data-test-id="content">
                Please enter your full UK postcode
            </span></p>
`;
